File: acmp_отбор/Contest 24.12.2025/Рунные слова.cpp
#include <bits/stdc++.h>

using namespace std;

void solve() {
    string s;
    if (!(cin >> s)) return;
    int n = s.length();
    int i = 0;
    if (n==1 or islower(s[0]) or isupper(s[n-1])) {
        cout << "No\n";
        return;
    }
    while (i < n) {
        if (isupper(s[i])) {
            int cnt = 1;
            i++;
            while (i < n && islower(s[i])) {
                i++;
                cnt++;
            }
            if (cnt < 2 or cnt > 4) {
                cout << "No\n";
                return;
            }
        }
        else {
            cout << "No\n";
            return;
        }
    }

    cout << "Yes\n";
}

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    solve();
    return 0;
}

File: acmp_отбор/Contest 24.12.2025/Рунные слова.py
def solve(s):
    i = 0
    n = len(s)
    if n==1 or s[0].islower() or s[-1].isupper():
        print("No")
        return
    while i < n:
        if s[i].isupper():
            cnt = 1
            i += 1
            while i < n and s[i].islower():
                i += 1
                cnt += 1
            if cnt < 2 or cnt > 4:
                print("No")
                return
        else:
            i += 1
    print("Yes")

s = input()
solve(s)
